app/presenters/chf/collection_show_presenter.rb

```diff
@@ -20,7 +20,8 @@ def needs_permission_badge?
     def urls_to_catalog
       return [] unless @solr_document.related_url.present?
       @urls_to_catalog ||= @solr_document.related_url.find_all do |url|
-        url.start_with?("http://othmerlib.chemheritage.org/record=") || url.start_with?("http://othmerlib.sciencehistory.org/record=")
+        url.include?("othmerlib.chemheritage.org/record=") ||
+        url.include?("othmerlib.sciencehistory.org/record=")
       end
```

Comment:
While this is _probably_ good enough, technically a URL _could_ include this string without it actually being the hostname. Probably unlikely, but say `http://example.org/othmerlib.chemheritage.org/record=/somethingelse` is technically a valid url. 

That's why i was doing the `starts_with`,  but I see that had a bug, repeating the same condition twice! It was meant to be `url.start_with?("http://othmerlib.chemheritage.org/record=") || url.start_with?("https://othmerlib.sciencehistory.org/record=")`  (the second one being https). 

Could also be expressed as a regexp `%r{\Ahttps?//othermerlib\.chemheritage\.org/record=}`.  Don't know if tha'ts more or less legible/prone to error. 

Do you think it's worth correcting to make sure it's only matching URLs that really are "start with" rather than the unlikely but possible ones like the above example?

Reply:
Absolutely. Will include a comment.

Reply:
oh wait, it was one sciencehistory and one chemheritage originally! I misread my own code, ha. Yeah, probably wroth changing to a regexp, or two regexps, if we want to make it more precise.

Reply:
Alternately, could be done with a `URI.parse` and checking hostname and path components independently (have to make sure to rescue parse errors). 

Or maybe it's good enough how it is, but it bothers me that it technically, if unlikely, could match a URL that it shouldn't. :)

Reply:
I think that's perfectly reasonable.